### PR TITLE
Add basic Vue support

### PR DESCRIPTION
**Pull request by cdrini**
Feature. Add support for Vue templates for non-essential Open Library experiences (i.e. browser support and non-js not as big a necessity).

Shared work between #2964 and #3788 

### Technical
Cool things:
- There's an easy way to setup an auto-reloading dev server 😍

Not so cool things:
- The size of our `all.js` JS file increased a little 😭 This is because we're now using core-js to auto-polyfilll things as they're imported.

Lots of things still to fix in future PRs:
- Enable vue i18n
- Vue is included in each web component; need to untangle that. Currently there isn't more than one per page, so won't be an issue.
- Vue is compiled in dev mode. _Definitely_ need to untangle that!
- Can't import pre-existing Less files into Vue

### Testing
- [x] /_dev/components loads the Vue component (twice), with buttons working
- [x] Rebased the merge UI against this branch and it works
- [x] Test on dev

### Screenshot

image

### Stakeholders
@mekarpeles

### CODE REVIEW

File: openlibrary/plugins/upstream/utils.py [hunk 1]
```diff
@@ -106,6 +106,38 @@ def render_template(name, *a, **kw):
         name = name.rsplit(".", 1)[0]
     return render[name](*a, **kw)
 
+
```
Comment (mekarpeles) [inline openlibrary/plugins/upstream/utils.py:109]: From https://travis-ci.org/github/internetarchive/openlibrary/jobs/729818889

```
0.62s$ if [ "$TRAVIS_PYTHON_VERSION" == "3.8" ]; then make lint-diff; fi
git diff master -U0 | ./scripts/flake8-diff.sh
openlibrary/plugins/upstream/utils.py:116:1: E303 too many blank lines (3)
openlibrary/plugins/upstream/utils.py:127:34: E225 missing whitespace around operator
openlibrary/plugins/upstream/utils.py:129:14: E261 at least two spaces before inline comment
openlibrary/plugins/upstream/utils.py:129:14: E262 inline comment should start with '# '
1     E225 missing whitespace around operator
1     E261 at least two spaces before inline comment
1     E262 inline comment should start with '# '
1     E303 too many blank lines (3)
make: *** [Makefile:72: lint-diff] Error 1
The command "if [ "$TRAVIS_PYTHON_VERSION" == "3.8" ]; then make lint-diff; fi" exited with 2.
```

File: Makefile [hunk 2]
```diff
@@ -36,6 +33,14 @@ js:
 		echo "\n// @license-end"  >> $$js; \
 	done
 
+components: $(COMPONENTS_DIR)/*.vue
+	mkdir --parents $(BUILD)
+	rm -rf $(BUILD)/components
+	for component in $^; do \
+		echo $$component; \
+		npx vue-cli-service build --mode development --no-clean --dest $(BUILD)/components --inline-vue --target wc --name "ol-$$(basename $$component .vue)" "$$component"; \
```
Comment (mekarpeles) [inline Makefile:41]: we discussed this may be problematic going forward

File: openlibrary/plugins/upstream/code.py [hunk 3]
```diff
@@ -48,6 +48,14 @@ class change_photo(change_cover):
 
 del delegate.modes['change_cover']     # delete change_cover mode added by openlibrary plugin
 
+
+class components_test(delegate.page):
+    path = "/_dev/components"
+
+    def GET(self):
+        return render_component('HelloWorld') + render_component('HelloWorld')
```
Comment (mekarpeles) [inline openlibrary/plugins/upstream/code.py:56]: strange that this generic endpoint is mapped to a single random HelloWorld component, but yolo

File: openlibrary/plugins/upstream/utils.py [hunk 4]
```diff
@@ -106,6 +106,38 @@ def render_template(name, *a, **kw):
...
+def kebab_case(upperCamelCase):
+    parts = re.findall(r'[A-Z][^A-Z]*', upperCamelCase)
+    return '-'.join(parts).lower()
+
+
+
+@public
+def render_component(name, attrs=None, json_encode=True):
```
Comment (mekarpeles) [inline openlibrary/plugins/upstream/utils.py:117]: @cdrini mentions `attrs` may have an issue (doesn't work)

File: openlibrary/plugins/upstream/utils.py [hunk 5]
```diff
@@ -106,6 +106,38 @@ def render_template(name, *a, **kw):
...
+    :param dict attrs: attributes to add to the component element
+    """
+    attrs = attrs or {}
+    attrs_str = ''
+    for (key, val) in attrs.items():
+        if json_encode and isinstance(val, dict) or isinstance(val, list):
+            val = simplejson.dumps(val)
+        attrs_str += ' %s="%s"' %(key, val.replace('"', "'"))
+
+    html = ''#'<script src="https://unpkg.com/vue@2.6.10"></script>'
```
Comment (mekarpeles) [inline openlibrary/plugins/upstream/utils.py:129]: remove confusing commment?

**Review by mekarpeles — APPROVED** [review]
lgtm, fix tests and feel free to merge

File: Makefile (hunk 2 above)
Comment (cdrini) [reply Makefile:41]: Added comment

File: openlibrary/plugins/upstream/code.py (hunk 3 above)
Comment (cdrini) [reply openlibrary/plugins/upstream/code.py:56]: Renamed to `components/HelloWorld`

File: openlibrary/plugins/upstream/utils.py (hunk 1 above)
Comment (cdrini) [reply openlibrary/plugins/upstream/utils.py:109]: Fixed

File: openlibrary/plugins/upstream/utils.py (hunk 5 above)
Comment (cdrini) [reply openlibrary/plugins/upstream/utils.py:129]: Fixed

File: openlibrary/plugins/upstream/utils.py (hunk 4 above)
Comment (cdrini) [reply openlibrary/plugins/upstream/utils.py:125]: Add comment to README

**Pull request merged**